src/service.rs

```diff
@@ -393,9 +403,77 @@ where
     Ok(Body::from_json(&f().await?)?.into())
 }
 
+async fn do_reset(mnemonic: SecretString, password: SecretString, config: &Config) -> State {
```

Comment:
Always prefer non-transient arguments first: 

```suggestion
async fn reset_state(config: &Config, mnemonic: SecretString, password: SecretString) -> State {
```